src/main/java/io/confluent/copycat/hdfs/FileUtils.java

```diff
@@ -38,19 +39,31 @@ public static void renameFile(final FileSystem fs, String sourcePath,
   }
 
   public static String tempFileName(String url, String topicsDir, TopicPartition topicPart) {
-    return fileName(url, topicsDir, topicPart, "tmp");
+    UUID id = UUID.randomUUID();
+    String name = id.toString() + "_" + "tmp";
```

Comment:
If we included the starting offset with the tempfile, would that let us do a better job of cleanup? If we knew the starting offset, it would be valid to try to delete a leftover tempfile once that offset had been committed via another file.